src/screens/Order/Orders.res

```diff
@@ -73,13 +79,87 @@ let make = (~previewOnly=false) => {
     }
   }
 
+  let updateViewsFilterValue = view => {
```

Comment:
This function might cause issue as you are `setfilterKeys` also then using the `filterKeys` I'm assuming both are useState return type or useRecoil. As both are async operations it can cause issue in updation. And if you have `setfilterKeys` why not to use that instead of push operation.